src/print/unicode.rs

```diff
@@ -137,79 +137,67 @@ pub fn print_unicode(graph: &GitGraph, settings: &Settings) -> Result<UnicodeGra
 
     // Compute branch lines in grid
     for (idx, info) in graph.commits.iter().enumerate() {
-        if let Some(trace) = info.branch_trace {
-            let branch = &graph.all_branches[trace];
-            let column = branch.visual.column.unwrap();
-            let idx_map = index_map[idx];
+        let Some(trace) = info.branch_trace else {
+            continue;
+        };
+        let branch = &graph.all_branches[trace];
+        let column = branch.visual.column.unwrap();
+        let idx_map = index_map[idx];
+
+        let branch_color = branch.visual.term_color;
+
+        grid.set(
+            column * 2,
+            idx_map,
+            if info.is_merge { CIRCLE } else { DOT },
+            branch_color,
+            branch.persistence,
+        );
+        for p in 0..2 {
+            let parent = info.parents[p];
+            let Some(par_oid) = parent else {
+                continue;
+            };
+            let Some(par_idx) = graph.indices.get(&par_oid) else {
+                // Parent is outside scope of graph.indices
+                // so draw a vertical line to the bottom
+                let idx_bottom = grid.height;
+                vline(
+                    &mut grid,
+                    (idx_map, idx_bottom),
+                    column,
+                    branch_color,
+                    branch.persistence,
+                );
+                continue;
+            };
+
+            let par_idx_map = index_map[*par_idx];
+            let par_info = &graph.commits[*par_idx];
+            let par_branch = &graph.all_branches[par_info.branch_trace.unwrap()];
+            let par_column = par_branch.visual.column.unwrap();
+
+            let (color, pers) = if info.is_merge {
+                (par_branch.visual.term_color, par_branch.persistence)
+            } else {
+                (branch_color, branch.persistence)
+            };
 
-            let branch_color = branch.visual.term_color;
+            if branch.visual.column == par_branch.visual.column {
+                if par_idx_map > idx_map + 1 {
+                    vline(&mut grid, (idx_map, par_idx_map), column, color, pers);
+                }
+            } else {
+                let split_index = super::get_deviate_index(graph, idx, *par_idx);
+                let split_idx_map = index_map[split_index];
+                let insert_idx = find_insert_idx(&inserts[&split_index], idx, *par_idx).unwrap();
+                let inx_split = split_idx_map + insert_idx;
```

Comment:
Is this a typo and should be `idx_split`?

Reply:
Yes. I will change it.